src/markov/markov_approx.jl

```diff
@@ -219,3 +223,567 @@ function estimate_mc_discrete{T}(X::Vector{T})
 
     return estimate_mc_discrete(X, states)
 end
+
+"""
+
+types specifying the method for `discrete_var`
+
+"""
+@compat abstract type AbstractMethod end
+immutable Even <: AbstractMethod end
+# immutable type Quantile <: AbstractMethod end
+# immutable type Quadrature <: AbstractMethod end
+
+doc"""
+
+Compute a finite-state Markov chain approximation to a VAR(1) process of the form
+
+```math
+    y_{t+1} = b + By_{t} + \Psi^{\frac{1}{2}}\epsilon_{t+1}
+```
+
+where ``\epsilon_{t+1}`` is an (M x 1) vector of independent standard normal
+innovations
+
+```julia
+P, X = discrete_var(b, B, Psi, Nm, nMoments, method, nSigmas)
+```
+
+##### Arguments
+
+- `b::ScalarOrArray` : (M x 1) constant vector
+                       (M=1 corresponds scalar case)
+- `B::ScalarOrArray` : (M x M) matrix of impact coefficients
+- `Psi::ScalarOrArray` : (M x M) variance-covariance matrix of the innovations
+- `Nm::Integer > 3` : Desired number of discrete points in each dimension
+
+##### Optional
+
+- `nMoments::Integer` : Desired number of moments to match. The default is 2.
+- `method::AbstractMethod` : AbstractMethod specifying the method used to determine
+              the grid points. Accepted inputs are `Even()`. Please see the
+             paper for more details.
+             NOTE: `Quantile()` and `Quadrature()` are not supported now.
+- `nSigmas::Real` : If the `Even()` option is specified, nSigmas is used to
+              determine the number of unconditional standard deviations
+              used to set the endpoints of the grid. The default is
+              sqrt(Nm-1).
+
+##### Returns
+
+- `P` : (Nm^M x Nm^M) probability transition matrix. Each row
+        corresponds to a discrete conditional probability
+        distribution over the state M-tuples in X
+- `X` : (M x Nm^M) matrix of states. Each column corresponds to an
+        M-tuple of values which correspond to the state associated
+        with each row of P
+
+##### NOTES
+
+- discrete_var only accepts non-singular variance-covariance matrices.
+- discrete_var only constructs tensor product grids where each dimension
+  contains the same number of points. For this reason it is recommended
+  that this code not be used for problems of more than about 4 or 5
+  dimensions due to curse of dimensionality issues.
+- Future updates will allow for singular variance-covariance matrices and
+  sparse grid specifications.
+
+"""
+function discrete_var{TI<:Integer}(b::ScalarOrArray, B::ScalarOrArray,
```

Comment:
The `TI` parameter doesn't seem necessary to me. The role it plays here is to restrict `Nm` and `nMoments` to have the same type. I don't think we need this.

What about writing it as `Nm::Integer, nMoments::Integer`.

I believe the only place `TI` is used in the body of the function is when defining the upper bound on the loop over `k`. I think we can probably just replace that with `Int` as it doesn't seem critical for that to match the type of `Nm` or `nMoments`